Sources/Commands/SwiftPackageCollectionsTool.swift

```diff
@@ -0,0 +1,358 @@
+            let license = optionalRow("License", version.license?.type.description)
+            let cves = optionalRow("CVEs", version.cves?.map { $0.identifier }.joined(separator: ", "))
+
+            return """
+                \(version.version)
+                Package Name: \(version.packageName)
+                Modules: \(modules)\(platforms)\(swiftVersions)\(license)\(cves)
+            """
+        }
+
+        mutating func run() throws {
+            guard let packageUrl = packageUrl else {
+                throw CollectionsError.missingArgument("packageUrl")
+            }
+
+            let identity = PackageReference.computeIdentity(packageURL: packageUrl)
+            let reference = PackageReference(identity: identity, path: packageUrl)
+
+            let result = try await { self.collections.getPackageMetadata(reference, profile: profileOptions.usedProfile, callback: $0) }
+
+            if let versionString = version {
+                guard let version = TSCUtility.Version(string: versionString), let result = result.package.versions.first(where: { $0.version == version }), let printedResult = printVersion(result) else {
+                    throw CollectionsError.invalidVersionString(versionString)
+                }
+
+                print("Version: \(printedResult)")
+            } else {
+                let description = optionalRow("Description", result.package.description)
+                let versions = result.package.versions.map { "\($0.version)" }.joined(separator: ", ")
+                let watchers = optionalRow("Watchers", result.package.watchersCount?.description)
+                let readme = optionalRow("Readme", result.package.readmeURL?.absoluteString)
+                let authors = optionalRow("Authors", result.package.authors?.map { $0.username }.joined(separator: ", "))
+                let latestVersion = optionalRow("--------------------------------------------------------------\nLatest Version", printVersion(result.package.latestVersion))
+
+                print("""
+                    \(description)Available Versions: \(versions)\(watchers)\(readme)\(authors)\(latestVersion)
+                """)
+            }
+        }
+    }
+}
+
+private func optionalRow(_ title: String, _ contents: String?) -> String {
+    if let contents = contents {
+        return "\n\(title): \(contents)\n"
+    } else {
+        return ""
+    }
+}
+
+private extension JSONEncoder {
+    func print<T>(_ value: T) throws where T : Encodable {
+        if #available(macOS 10.13, *) {
+            self.outputFormatting = [.prettyPrinted, .sortedKeys]
+        }
+
+        let jsonData = try self.encode(value)
+        let jsonString = String(data: jsonData, encoding: .utf8)!
+        Swift.print(jsonString)
+    }
+}
+
+private extension ParsableCommand {
+    var collections: PackageCollectionsProtocol {
+        fatalError("not implemented")
```

Comment:
We have to figure out how to get the necessary configuration and how to choose the storage implementation etc.

Reply:
cc @abertelrud: this is the practical aspect of the question I was asking about the DI pattern we follow in SwiftPM. If there is none, we could just create the "correct" instances here 

=> Configuration should have a default ctor that will read the default config: 
=> Storage we want to use the SQLite and File implementation , make sure to close the storage when done!

Reply:
If there is a pattern in SwiftPM, I haven't found it.  So I agree, we should establish one and document it.

> => Configuration should have a default ctor that will read the default config:
> => Storage we want to use the SQLite and File implementation , make sure to close the storage when done!

I'm not sure how to translate that to what the API would look like in this case.  What component would make the choice of storage backend in this case?

Reply:
the api would look the same. the call site will need to make the decision on what concrete impls of configuration and storage to use, but we would include "default impls" which would be the natural choice. we can also add an empty ctor that uses these default impls if we think its worth it

Reply:
I think an empty initializer would be nice, because then clients don't need to explicitly know about things like the SQLite implementation etc.